src/cluster_legacy.c

```diff
@@ -3128,63 +3128,38 @@ int clusterProcessPacket(clusterLink *link) {
 
         /* Many checks are only needed if the set of served slots this
          * instance claims is different compared to the set of slots we have
-         * for it. Check this ASAP to avoid other computational expansive
-         * checks later. */
-        clusterNode *sender_primary = NULL; /* Sender or its primary if replica. */
-        int dirty_slots = 0;                /* Sender claimed slots don't match my view? */
-
-        if (sender) {
-            sender_primary = clusterNodeIsPrimary(sender) ? sender : sender->replicaof;
-            if (sender_primary) {
-                dirty_slots = memcmp(sender_primary->slots, hdr->myslots, sizeof(hdr->myslots)) != 0;
```

Comment:
So if a node is a replica, it still reports its slots and we used this to update the slots for its primary? And we don't want to do that anymore?

Reply:
>So if a node is a replica, it still reports its slots and we used this to update the slots for its primary? 

No - this was the regression I introduced in #445. 

>And we don't want to do that anymore?

Yup